Remotion/Web/ClientScript/Scripts/TreeView.ts

```diff
@@ -100,19 +98,22 @@ class WebTreeView
           event.preventDefault();
           event.stopPropagation();
 
-          let hasChildren = $activeTreeNode.attr ('aria-expanded') === 'true';
+          let hasChildren = activeTreeNode.getAttribute ('aria-expanded') === 'true';
           if (!hasChildren)
           {
-            var $parentTreeNode = $activeTreeNode.parentsUntil ('ul[role=tree]', 'li').first();
-            if ($parentTreeNode.length > 0)
+            let parent = activeTreeNode.parentElement;
+            while (parent && !parent.matches ('ul[role=tree],li'))
```

Comment:
We stop when we find either UL element with role tree (the container) or the LI element (the tree node). Since we only care about the first LI element we encounter this works correctly. 

This is also why we need to check for LI some lines done. There are three ways we can leave this loop:
 - `parent === null` because we have hit the root node for some reason
 - `element is LI` because we have hit a tree node
 - `element is UL` because we have hit the container

Reply:
That's true. Do we have a way to better express the semantics while still keeping the code from getting bloated? I'm looking for way to express the intention you specified without having to think about the code while reading it, if that's possible.

Reply:
I don't think that there is a good way without bloating the loop and increasing the complexity. We could put the result object into its own variable and assign it inside of the loop, this way we split the exit conditions from the success condition at the cost of a more complex loop.

Reply:
mhm, it would be great if you could paste a comment with the current implementation and the alternative as code blocks. then we can see how each actually feels and decide then.

Reply:
Here are some possible solutions. It would also be possible to convert the while loop into a for loop

```ts
function currentSolution(activeTreeNode: HTMLElement)
{
    let parent = activeTreeNode.parentElement;
    while (parent && !parent.matches ('ul[role=tree],li'))
        parent = parent.parentElement;

    if (parent && parent.tagName.toLowerCase() === 'li')
    {
        let newTreeNode = parent;
        WebTreeView.SetFocus (newTreeNode, activeTreeNode);
    }
}

function separateListItemVariable(activeTreeNode: HTMLElement)
{
    let target: HTMLElement | null = null;
    let parent = activeTreeNode.parentElement;
    while (parent && !parent.matches ('ul[role=tree]'))
    {
        if (parent.tagName.toLowerCase() === 'li')
        {
            target = parent;
            break;
        }

        parent = parent.parentElement;
    }

    if (target)
    {
        let newTreeNode = parent;
        WebTreeView.SetFocus (newTreeNode, activeTreeNode);
    }
}

function actionInsideTheLoop(activeTreeNode: HTMLElement)
{
    let parent = activeTreeNode.parentElement;
    while (parent && !parent.matches ('ul[role=tree]'))
    {
        if (parent.tagName.toLowerCase() === 'li')
        {
            let newTreeNode = parent;
            WebTreeView.SetFocus (newTreeNode, activeTreeNode);
            break;
        }

        parent = parent.parentElement;
    }
}
```

Reply:
Hmm... yes, personally I try to avoid `break` because it makes the flow harder to read but in this particular case, I think the last option actionInsideTheLoop (with or without a for instead of a while) is the one that's the easiest to reason over. You have the guarding stop condition in the loop condition and the "hit" stop condition as a dedicated piece of code. This means, you can focus on each aspect individually.

Worth changing the code?